build.gradle

```diff
@@ -203,6 +202,61 @@ task wrapper(type: Wrapper) { gradleVersion = '1.12' }
 
 import javax.tools.ToolProvider
 
+task javadocTarball(type: Tar) {
+  baseName = "gobblin-javadoc-all"
+  destinationDir = new File(project.buildDir, baseName)
+  compression = Compression.GZIP
+  extension = 'tgz'
+  description = "Generates a tar-ball with all javadocs to ${destinationDir}/${archiveName}"
+}
+
+javadocTarball << {
+  def indexFile = new File(destinationDir, "index.md")
+  def version = rootProject.ext.javadocVersion
+  indexFile << """----
+layout: page
+title: Gobblin Javadoc packages ${version}
+permalink: /javadoc/${version}/
+----
+
+"""
+  rootProject.ext.javadocPackages.each {
+    indexFile << "* [${it}](${it})\n"
+  }
+}
+
+//Javadoc initialization for subprojects
+ext.javadocVersion = null != project.version ? project.version : "latest"
+if (ext.javadocVersion.indexOf('-') > 0) {
+  //remove any "-" addons from the version
+  ext.javadocVersion = ext.javadocVersion.substring(0, ext.javadocVersion.indexOf('-'))
+}
+
+ext.javadocPackages = new HashSet<String>()
+subprojects.each{Project pr -> 
+  if (file(pr.projectDir.absolutePath + "/src/main/java").exists()) {
+    rootProject.ext.javadocPackages += pr.name
+  }
+}
+ext.javadocLinks = ext.javadocPackages.collect{"http://linkedin.github.io/gobblin/javadoc/${ext.javadocVersion}/${it}/"}
+//println "GOBBLIN_LINKS=" + ext.javadocLinks
+
+subprojects {
```

Comment:
Yes, I find it harder to read when we have one ginormous subprojects {} block. Further, "subproject {} " (configuration) is different from "subproject.each {}" (iteration)  and those get evaluated at different times.